homeassistant/components/climate/radiotherm.py

```diff
@@ -61,35 +80,31 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     tstats = []
 
     for host in hosts:
-        try:
-            tstat = radiotherm.get_thermostat(host)
-            tstats.append(RadioThermostat(tstat, hold_temp, away_temps))
-        except OSError:
-            _LOGGER.exception("Unable to connect to Radio Thermostat: %s",
-                              host)
+        tstats.append(RadioThermostat(host, hold_temp, away_temps))
```

Comment:
That is not realy a improvment. You porting stuff out from library into home-assistant. Instead you make a hyprid of library/hass code base, update the `radiotherm` and remove the post_url function